homeassistant/components/seitron/__init__.py

```diff
@@ -0,0 +1,70 @@
+"""The seitron integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+import async_timeout
+from pyseitron.seitron_gateway import SeitronGateway
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import aiohttp_client, config_entry_oauth2_flow
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from . import api
+from .const import DOMAIN  # seitron
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS: list[Platform] = [Platform.CLIMATE]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up seitron from a config entry."""
+    _LOGGER.info("Init: setup: EntryId %s", entry.entry_id)
+
+    async def _async_update_data():
+        """Update data via library."""
+        _LOGGER.info("Init: Coordinator callback")
+        async with async_timeout.timeout(20):
+            await seitron_gw.update()
+        return seitron_gw
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name=DOMAIN.title(),
+        update_method=_async_update_data,
+        update_interval=timedelta(seconds=60),
+    )
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = coordinator
```

Comment:
Don't store it until `async_config_entry_first_refresh` was successful.

Reply:
fixed